src/transaction.rs

```diff
@@ -151,6 +151,7 @@ impl Config {
 pub struct Transaction<'conn> {
     conn: &'conn Connection,
     depth: u32,
+    savepoint_name: Option<&'conn str>,
```

Comment:
I'd just make this an `Option<String>` - the runtime cost of the allocation is ~0 compared to talking over the network.